pkg/controller/controller_instance.go

```diff
@@ -513,19 +513,9 @@ func (c *controller) reconcileServiceInstance(instance *v1beta1.ServiceInstance)
 		return nil
 	}
 
-	// If there's no async op in progress, determine whether there is a new
-	// generation of the object. If the instance's generation does not match
-	// the reconciled generation, then there is a new generation, indicating
-	// that changes have been made to the instance's spec. If there is an
-	// async op in progress, we need to keep polling, hence do not bail if
-	// there is not a new generation.
-	//
-	// Note: currently the instance spec is immutable because we do not yet
-	// support plan or parameter updates.  This logic is currently meant only
-	// to facilitate re-trying provision requests where there was a problem
-	// communicating with the broker.  In the future the same logic will
-	// result in an instance that requires update being processed by the
-	// controller.
+	// If the instance's generation does not match the reconciled
+	// generation, then there is a new generation, indicating that changes
+	// have been made to the instance's spec.
```

Comment:
If we're going to tweak the comment to match the code, let's word the comment to align with the if-statement, so perhaps:
```
// If the instance's "generation" matches the "reconciled generation", then
// the "generation" has not changed and that means no new changes
// have been made to the instance's spec.
```

Reply:
I'm +1 on doug's suggestions, except that i would prefer to use jsonpaths to describe the fields like `metadata.generation` and `status.reconciledGeneration`

Reply:
Done.